app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(animal: Animal) -> Animal:
+        if isinstance(animal, Herbivore) and not animal.hidden:
+            animal.health -= 50
+        if animal.health <= 0:
+            Animal.alive.remove(animal)
```

Comment:
logic is telling us that we either bite animal and remove it from list, or it should be removed already. I mean it should adhere to bite method, not to overall removage, since we could have some rare occasions, where we won't want some object to be removed before biting, but it could be <1 health.